grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandlers.java

```diff
@@ -101,160 +43,13 @@ static UnframedGrpcErrorHandler of(
-            final Throwable cause = responseCause(ctx);
-            final HttpStatus httpStatus = mappingFunction.apply(ctx, status, cause);
-            final HttpHeaders trailers = !response.trailers().isEmpty() ?
-                                         response.trailers() : response.headers();
-            final String grpcStatusDetailsBin = trailers.get(GrpcHeaderNames.GRPC_STATUS_DETAILS_BIN);
-            final ResponseHeaders responseHeaders = ResponseHeaders.builder(httpStatus)
-                                                                   .contentType(MediaType.JSON_UTF_8)
-                                                                   .addInt(GrpcHeaderNames.GRPC_STATUS,
-                                                                           grpcCode.value())
-                                                                   .build();
-            boolean success = false;
-            try (OutputStream outputStream = new ByteBufOutputStream(buffer);
-                 JsonGenerator jsonGenerator = mapper.createGenerator(outputStream)) {
-                jsonGenerator.writeStartObject();
-                jsonGenerator.writeNumberField("code", grpcCode.value());
-                jsonGenerator.writeStringField("grpc-code", grpcCode.name());
-                if (grpcMessage != null) {
-                    jsonGenerator.writeStringField("message", grpcMessage);
-                }
-                if (cause != null && ctx.config().verboseResponses()) {
-                    jsonGenerator.writeStringField("stack-trace", Exceptions.traceText(cause));
-                }
-                if (!Strings.isNullOrEmpty(grpcStatusDetailsBin)) {
-                    com.google.rpc.Status rpcStatus = null;
-                    try {
-                        rpcStatus = decodeGrpcStatusDetailsBin(grpcStatusDetailsBin);
-                    } catch (InvalidProtocolBufferException e) {
-                        logger.warn("Unexpected exception while decoding grpc-status-details-bin: {}",
-                                    grpcStatusDetailsBin, e);
-                    }
-                    if (rpcStatus != null) {
-                        jsonGenerator.writeFieldName("details");
-                        writeErrorDetails(rpcStatus.getDetailsList(), jsonGenerator, jsonMarshaller);
-                    }
-                }
-                jsonGenerator.writeEndObject();
-                jsonGenerator.flush();
-                success = true;
-            } catch (IOException e) {
-                logger.warn("Unexpected exception while generating a JSON response", e);
-            } finally {
-                if (!success) {
-                    buffer.release();
-                }
-            }
-            if (success) {
-                return HttpResponse.of(responseHeaders, HttpData.wrap(buffer));
+                return JsonUnframedGrpcErrorHandler.of(mappingFunction, jsonMarshaller)
```

Comment:
This is not your fault but this may create `JsonUnframedGrpcErrorHandler` or `TextUnframedGrpcErrorHandler` whenever this method is called.

Should we add `DefaultUnframedGrpcErrorHandler` that has `JsonUnframedGrpcErrorHandler` and `TextUnframedGrpcErrorHandler` as member fields?
```java
final class DefaultUnframedGrpcErrorHandler implements UnframedGrpcErrorHandler {
  
  private final JsonUnframedGrpcErrorHandler jsonUnframedGrpcErrorHandler;
  private final TextUnframedGrpcErrorHandler textUnframedGrpcErrorHandler;
  ...
}
```

Reply:
We can delete `UnframedGrpcErrorHandlers` class then 👍